x-pack/plugins/security_solution/public/management/pages/blocklist/view/components/blocklist_form.tsx

```diff
@@ -114,10 +114,8 @@ function isValid(itemValidation: ItemValidation): boolean {
 // eslint-disable-next-line react/display-name
 export const BlockListForm = memo<ArtifactFormComponentProps>(
   ({ item, policies, policiesIsLoading, onChange, mode }) => {
-    const [visited, setVisited] = useState<{ name: boolean; value: boolean }>({
-      name: false,
-      value: false,
-    });
+    const [nameVisited, setNameVisited] = useState(false);
+    const [valueVisited, setValueVisited] = useState({ value: false }); // Use object to trigger re-render
```

Comment:
So take this with a grain of salt 🧂 because I haven't really touched frontend code in nearly 3 years but this feels a little bit weird. Not a blocker or anything or I do wonder if there's a better approach.

Reply:
i agree with joey here, this doesn't look like a good practice.

i guess it is needed, because warnings and errors are stored in reference values, but i don't see why we need to store those in refs instead of good ol' states. i'd suggest to try to remove the refs, and use states (or maybe one state if they always change together) to improve this file